src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -84,6 +104,87 @@ static bool DecodeEntry(zcbor_state_t * states, void * buffer, size_t bufferSize
     return res;
 }
 
+bool SerializeFactoryData(uint8_t * factoryDataBuff, size_t factoryDataSize, const struct FactoryData * factoryData)
+{
+    zcbor_state_t cbor_state[2];
+    zcbor_new_encode_state(cbor_state, MAX_FACTORY_DATA_NESTING_LEVEL, factoryDataBuff, factoryDataSize, 0);
+
+    bool res = zcbor_map_start_encode(cbor_state, 1);
+
+    res = res && zcbor_tstr_put_lit_cast(cbor_state, "version");
```

Comment:
The result of this method will be true only if all of the below operations succeed, right? I think it would be reasonable to return after the first failure, as executing dozen of the following operations is pointless.